components/lib/crash/src/main/java/mozilla/components/lib/crash/handler/CrashHandlerService.kt

```diff
@@ -29,10 +31,18 @@ class CrashHandlerService : IntentService(WORKER_THREAD_NAME) {
 
         intent.extras?.let { extras ->
             val crash = Crash.NativeCodeCrash.fromBundle(extras)
+            val skipPrompt = extras.getBoolean(Crash.INTENT_EXTRA_SKIP_PROMPT, false)
+
+            if (skipPrompt) {
```

Comment:
I'm concerned about reusing the `CrashHandlerService` class here: It's main purpose is to receive crashes from GeckoView (hence the `Crash.NativeCodeCrash.fromBundle()` above) and forward it to the `CrashReporter` instance.

With this patch it also receives crashes from the notification (which may not be `Crash.NativeCodeCrash`!). Since we kill this process immediately I wonder if we'd even be able to report any crashes (which may happen asynchronously).

We may want to consider using `CrashReporter.submitReport()` (like `CrashReporterActivity` does) and wait for it to completely. Not sure from where though? Maybe a foreground service? Or a workmanager worker (which would still need a service to schedule it?)? A workmanager worker would have the benefit that we could fix some of our other lib-crash issues (e.g. CrashReporterActivity blocking, CrashReporterActivity trying only once, ..). What do you think?

Reply:
I see, I've misunderstood the role of CrashHandlerService then.  I'll look into both the foreground service and the worker idea.  Thanks,